pallets/pallet-web3-names/src/web3_name.rs

```diff
@@ -0,0 +1,152 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_std::{fmt::Debug, marker::PhantomData, str, vec::Vec};
+
+use codec::{Decode, Encode};
+use frame_support::{ensure, sp_runtime::SaturatedConversion, traits::Get, BoundedVec};
+use scale_info::TypeInfo;
+
+use crate::{Config, Error};
+
+/// A KILT web3 name.
+///
+/// It is bounded in size (inclusive range [MinLength, MaxLength]) and can only
+/// contain a subset of ASCII characters.
+#[derive(Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T, MaxLength, MinLength))]
+pub struct AsciiWeb3Name<T, MinLength, MaxLength>(pub(crate) BoundedVec<u8, MaxLength>, PhantomData<(T, MinLength)>);
+
+impl<T: Config> TryFrom<Vec<u8>> for AsciiWeb3Name<T, T::MinNameLength, T::MaxNameLength> {
+	type Error = Error<T>;
+
+	/// Fallible initialization from a provided byte vector if it is below the
+	/// minimum or exceeds the maximum allowed length or contains invalid ASCII
+	/// characters.
+	fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+		ensure!(
+			value.len() >= T::MinNameLength::get().saturated_into(),
+			Self::Error::Web3NameTooShort
+		);
+		let bounded_vec: BoundedVec<u8, T::MaxNameLength> =
+			BoundedVec::try_from(value).map_err(|_| Self::Error::Web3NameTooLong)?;
+		ensure!(
+			is_byte_array_ascii_string(&bounded_vec),
+			Self::Error::InvalidWeb3NameCharacter
+		);
+		Ok(Self(bounded_vec, PhantomData))
+	}
+}
+
+/// Verify that a given slice contains only allowed ASCII characters.
+fn is_byte_array_ascii_string(input: &[u8]) -> bool {
```

Comment:
I would not specify the input type in the function name. Also it's not checking if it's ascii. The condition is much stricter.

```suggestion
fn is_valid_web3_name(input: &[u8]) -> bool {
```

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/314/commits/6ab13cfb0b2747b9e18b6a2172df6171e9972c94